fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+            validateRecord(store, result.get(recordNumber), FDBRecordStoreBase.RecordValidationOptions.DESERIALIZABLE);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingVersion [splitLongRecords = {0}]")
+    @BooleanSource
+    public void testValidateRecordsMissingVersion(boolean splitLongRecords) throws Exception {
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION, hook);
+        // Delete the versions
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+            Tuple pkAndVersion = result.get(0).getPrimaryKey().add(-1);
+            byte[] versionKey = store.recordsSubspace().pack(pkAndVersion);
+            store.ensureContextActive().clear(versionKey);
+
+            pkAndVersion = result.get(1).getPrimaryKey().add(-1);
+            versionKey = store.recordsSubspace().pack(pkAndVersion);
+            store.ensureContextActive().clear(versionKey);
+            commit(context);
+        }
+
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+            validateRecord(store, result.get(0), FDBRecordStoreBase.RecordValidationOptions.VALID_VERSION);
+            validateRecord(store, result.get(1), FDBRecordStoreBase.RecordValidationOptions.VALID_VERSION);
+            commit(context);
+        }
+    }
+
+    @Nonnull
+    private static RecordMetaDataHook getRecordMetaDataHook(final boolean splitLongRecords) {
+        return metaData -> {
+            metaData.setSplitLongRecords(splitLongRecords);
+            // index cannot be used with large fields
+            metaData.removeIndex("MySimpleRecord$str_value_indexed");
+        };
+    }
+
+    private void validateRecord(FDBRecordStore store, final FDBStoredRecord<Message> rec, final FDBRecordStoreBase.RecordValidationOptions validationOption) throws Exception {
+        EnumSet<FDBRecordStoreBase.RecordValidationOptions> recordValidationOptions;
+        recordValidationOptions = store.validateRecordAsync(rec.getPrimaryKey(), EnumSet.allOf(FDBRecordStoreBase.RecordValidationOptions.class), false).get();
+        assertNotNull(recordValidationOptions);
+        EnumSet<FDBRecordStoreBase.RecordValidationOptions> expected = (validationOption == null) ?
+                                                                       EnumSet.noneOf(FDBRecordStoreBase.RecordValidationOptions.class) :
+                                                                       EnumSet.of(validationOption);
```

Comment:
Are there situations where it could return more than one element?
Would it be better to have it return `@Nullable RecordValidationOptions`?

Reply:
This API has changed